org.adempiere.base/src/org/compiere/model/MProductionLineMA.java

```diff
@@ -99,17 +100,24 @@ public static MProductionLineMA get( MProductionLine parent, int asi, Timestamp
 	@Override
 	protected boolean beforeSave(boolean newRecord) 
 	{
-		MProductionLine parentLine = new MProductionLine(getCtx(), getM_ProductionLine_ID(), get_TrxName());
-		MProduction prodParent;
-		if (parentLine.getM_Production_ID() > 0) {
-			prodParent = new MProduction(getCtx(), parentLine.getM_Production_ID(), get_TrxName());
-		} else {
-			MProductionPlan plan = new MProductionPlan(getCtx(), parentLine.getM_ProductionPlan_ID(), get_TrxName());
-			prodParent = new MProduction(getCtx(), plan.getM_Production_ID(), get_TrxName());
-		}
-		if (newRecord && prodParent.isProcessed()) {
-			log.saveError("ParentComplete", Msg.translate(getCtx(), "M_Production_ID"));
-			return false;
+		if(newRecord) {
+			String sql = "SELECT 1 FROM M_Production "
+					+ "WHERE Processed = 'Y' AND M_Production_ID IN"
+					+ " (SELECT M_Production_ID FROM M_ProductionLine WHERE M_ProductionLine_ID = ?)";

```

Comment:
I think it is more natural and efficient that this should be changed to M_Production Inner Join M_ProductionLine instead.

Reply:
Done